src/renderer/components/+workspaces/workspace-menu.tsx

```diff
@@ -17,6 +20,16 @@ interface Props extends Partial<MenuProps> {
 export class WorkspaceMenu extends React.Component<Props> {
   @observable menuVisible = false;
 
+  activateWorkspace = (id: WorkspaceId) => {
+    const clusterId = workspaceStore.getById(id).lastActiveClusterId;
+    workspaceStore.setActive(id);
+    if (clusterId) {
+      navigate(clusterViewURL({ params: { clusterId } }));
+    } else {
+      navigate(landingURL());
+    }
```

Comment:
Why is this code different from the code in `workspaces.tsx`?

Reply:
When you hit on workspace link from Workspaces page, there's no redirect to dashboard - user will stay on same page and see that `current` indicator will jump to clicked workspace.